### ISSUE TITLE
install_libs.sh fails

### ISSUE THREAD

**Issue by ZXMushroom63** [Rain92/kobo-qt-setup-scripts#5]
The install_libs.sh script fails immediately, with this error:

~/kobo-qt-setup-scripts$ ./install_libs.sh
Cloning into 'zlib-ng'...
remote: Enumerating objects: 17050, done.
remote: Counting objects: 100% (2107/2107), done.
remote: Compressing objects: 100% (524/524), done.
remote: Total 17050 (delta 1667), reused 1695 (delta 1583), pack-reused 14943
Receiving objects: 100% (17050/17050), 9.44 MiB | 8.36 MiB/s, done.
Resolving deltas: 100% (12133/12133), done.
HEAD is now at 6ff8b52c Support RVV hwcap detect at runtime
error: patch failed: arch/arm/arm_features.c:1
error: arch/arm/arm_features.c: patch does not apply

**Comment by ZXMushroom63** [comment]
I was doing this in windows subsystem for linux, I will try in a linux VM and see if that works.

**Comment by ZXMushroom63** [comment]
Right, managed to fix this issue in my own fork: https://github.com/ZXMushroom63/kobo-qt-setup-scripts

**Comment by Aryetis** [comment]
Not sure how far you got on your attempt to fix it. But your fork didn't do it for me, it still crashed during the libraries stage. Thanks for trying tho.

On my end I've managed to fix every single issue I had with it and can finally get back on working on my Kobo app. Here's my fork of the repository https://github.com/Aryetis/kobo-qt-setup-scripts  . I'll write a PR for it later.

Not flexing but contrary to you I didn't use sudo every step of the way, didn't swap libs's repository and managed to compile all of them. Thus keeping the same qt config as the original. It might help you, Also it works just fine on my WSL2 debian. Here's more instructions on how I use it https://github.com/Aryetis/Obenkyobo/blob/master/DevReadme.md . I just fixed it today and I'm kinda sick so expect this repository and readme to have a lot of minor commit in the following hours/days.

**Comment by ZXMushroom63** [comment]
I fixed one issue, there were more hiding behing it. And, I have no C++ experience whatsoever, so it was pretty much fumbling in the dark to see what worked :P

I'll check out your fork, becuase I've been unable to write apps for my Clara HD unless it is in a half-baked build of Qt 4, which is not optimal.

Thanks!

**Comment by ZXMushroom63** [comment]
First bits seem to work flawlessly, though you should probably put in the readme that you need to use --recurse-submodules on git clone, it got me confused for a while.

**Comment by Aryetis** [comment]
I haven't touched kobo-qt-setup-scripts's readme yet. If my PR get merged it would need to be updated anyways as I ve modified the way the deploy script works too.
But I ve reflected all those changes ( and included --recurse-submodules instructions ) in Obenkyobo's DevReadme.

The last issue I have at the moment is that when debugging my code, QtCreator have issues using gdb's Debugger Helpers. It could be a QtCreator bug tho, I still have to investigate properly but I m kinda sick at the moment so it might take a few days until I figure it out.

EDIT : Turns out as I suspected the debug problem is a QtCreator v9 bug .... It should be fixed in QtCreator V10. So if like me you re using debian wsl2, you can either; compile a newer QtCreator; fix it yourself cf https://microhobby.com.br/blog/2023/08/08/qt-creator-v9-0-2-debug-helpers-not-working-on-ubuntu-23-04-and-debian-bookworm-12/ or get a binary any other way.
On my end I'll probably update to Debian Trixie (testing) as its qtcreator package's version is 10.0.2-4.
EDIT 2 : updating to Trixie brings other problems with the toolchain. I recommend sticking to Bookworm for now and either patching QtCreator or installing a new one from offline installers.
